pkg/controller/volume/expand/cache/volume_resize_map.go

```diff
@@ -195,15 +211,13 @@ func (resizeMap *volumeResizeMap) UpdatePVSize(pvcr *PVCWithResizeRequest, newSi
 }
 
 func (resizeMap *volumeResizeMap) updatePVCCapacityAndConditions(pvcr *PVCWithResizeRequest, newSize resource.Quantity, pvcConditions []v1.PersistentVolumeClaimCondition) error {
-	claimClone := pvcr.PVC.DeepCopy()
-
-	claimClone.Status.Capacity[v1.ResourceStorage] = newSize
-	claimClone.Status.Conditions = pvcConditions
-
-	_, updateErr := resizeMap.kubeClient.CoreV1().PersistentVolumeClaims(claimClone.Namespace).UpdateStatus(claimClone)
-	if updateErr != nil {
-		glog.V(4).Infof("updating PersistentVolumeClaim[%s] status: failed: %v", pvcr.QualifiedName(), updateErr)
-		return updateErr
-	}
-	return nil
+	_, err := util.UpdatePVCCondition(
+		pvcr.PVC,
+		pvcConditions,
+		resizeMap.kubeClient,
+		func(newPVC *v1.PersistentVolumeClaim) *v1.PersistentVolumeClaim {
```

Comment:
Why indirection? It makes the code hard to read. IMO, it would be more readable to have something like:

```go
newPVC := pvcr.PVC.DeepCopy()
newPVC.Status.Capacity[v1.ResourceStorage] = newSize
UpdateCondition(newPVC, conditionName, newValue)
err := PatchPVC(resizeMap.kubeClient, pvcr.PVC /*old PVC*/, newPVC)
```

We could use PatchPVC everywhere we update PVCs

Reply:
fixed.